test/framework/src/main/java/org/elasticsearch/transport/nio/ESSelector.java

```diff
@@ -46,12 +46,12 @@
 
     final Selector selector;
     final ConcurrentLinkedQueue<NioChannel> channelsToClose = new ConcurrentLinkedQueue<>();
-    final Set<NioChannel> registeredChannels = Collections.newSetFromMap(new ConcurrentHashMap<NioChannel, Boolean>());
 
     private final EventHandler eventHandler;
     private final ReentrantLock runLock = new ReentrantLock();
     private final AtomicBoolean isClosed = new AtomicBoolean(false);
     private final PlainActionFuture<Boolean> isRunningFuture = PlainActionFuture.newFuture();
+    private final Set<NioChannel> registeredChannels = Collections.newSetFromMap(new ConcurrentHashMap<NioChannel, Boolean>());
```

Comment:
I wonder if we can simply register `Closeables` here instead, then the `NIOChannel` can simply impl it and we can do whatever happens in `eventHandler.handleClose(channel);` in the close method?